src/utils/split-props.ts

```diff
@@ -0,0 +1,32 @@
+type Omit<T extends Dictionary<any>, K extends keyof T> = Pick<T, Exclude<keyof T, K>>
+
+interface Dictionary<T> {
+  [key: string]: T
+}
+
+export interface SplitProps<P, K extends keyof P> {
+  matchedProps: Pick<P, K>
+  remainingProps: Omit<P, K> // Omit<P, K>
```

Comment:
Any reason we add `Omit<P, K>` as a comment again? 😆

Reply:
Oh, leftover comment. I'll remove.